src/6.2 - Decorator with param.py

```
@@ -0,0 +1,42 @@
+"""
+Создайте декоратор, который вызывает задекорированную функцию пока она
+не выполнится без исключений (но не более n раз - параметр декоратора).
+Если превышено количество попыток, должно быть возбуждено исключение
+типа TooManyErrors.
+"""
+
+
+class TooManyErrors(Exception):
+    pass
+
+
+def my_dec(times):
+
+    def dec(func):
+        nonlocal times
+
+        def wrapper(*args, **kwargs):
+            nonlocal times
+            res = 0
+            try:
+                if not times:
+                    raise TooManyErrors()
+                res = func(*args, **kwargs)
+                print(res)
+                times -= 1
+            except TooManyErrors:
```

why do we suppress this exception inside decorator?

---

you catch only one exception here and should catch all